nezha: 0.20.3 -> 1.9.5; nezha-agent: 0.20.5 -> 1.9.2; nixos/nezha-agent: refactor

Moraxyc opened the pull request:
BREAKING CHANGE: Incompatible Configuration Changes
- The upgrade includes significant architectural changes, resulting in a complete incompatibility with previous configurations.
- Users will need to reconfigure their setups as the old configuration is no longer supported.
- Please review the updated documentation for the new configuration format and features.
Docs: https://nezha.wiki

Due to changes in the communication protocol between the client and server, the agent‘s upgrade also be included in this PR.

~~Must be merged after #374097~~(Done)
~~Must be merged after #364131 and #364145~~(Done)

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

definfo commented on pkgs/by-name/ne/nezha/adminFrontend.nix:
```diff
@@ -0,0 +1,40 @@
+  src = fetchFromGitHub {
+    owner = "nezhahq";
+    repo = "admin-frontend";
+    rev = "refs/tags/v${version}";
```
```suggestion
    rev = "v${version}";
```

Moraxyc replied:
Could you please provide some references? As I understand, the `refs/tags/` path is used to limit the scope to the specific tag, but not the same name branch.

definfo replied:
https://ryantm.github.io/nixpkgs/builders/fetchers/#fetchfromgithub
> `rev` corresponds to the Git commit hash or tag (e.g v1.0) that will be downloaded from Git.

Moraxyc committed:
nezha: 0.20.3 -> 1.9.5

Diff: https://github.com/nezhahq/nezha/compare/refs/tags/v0.20.3...v1.9.5

BREAKING CHANGE: Incompatible Configuration Changes
- The upgrade includes significant architectural changes, resulting in a complete incompatibility with previous configurations.
- Users will need to reconfigure their setups as the old configuration is no longer supported.
- Please review the updated documentation for the new configuration format and features.

Docs: https://nezha.wiki

Moraxyc committed:
nezha-agent: 0.20.5 -> 1.9.2

Diff: https://github.com/nezhahq/agent/compare/refs/tags/v0.20.5...v1.9.2

BREAKING CHANGE: Incompatible Configuration Changes
- The upgrade includes significant architectural changes, resulting in a complete incompatibility with previous configurations.
- Users will need to reconfigure their setups as the old configuration is no longer supported.
- Please review the updated documentation for the new configuration format and features.

Docs: https://nezha.wiki

Moraxyc committed:
nixos/nezha-agent: refactor

Moraxyc committed:
nixos/doc/rl-2505: mention Nezha update

wolfgangwalther reviewed (commented):
Not a review, just spotted this randomly.

Moraxyc commented on pkgs/by-name/ne/nezha/package.nix:
```diff
@@ -2,25 +2,84 @@
-  version = "0.20.13";
+  version = "1.5.6";
+
+  fontendName = lib.removePrefix "nezha-theme-";
```
aha, a little typo
thx to my nix language server, autocompleting helps me use it in whole file

Moraxyc commented:
> #370824 is merged, can you rebase on latest master to confirm that Eval / Tag will run correctly?

https://github.com/NixOS/nixpkgs/actions/runs/12610251767/job/35144742184?pr=361515

Thanks! It works good now.

L-Trump commented:
nezha v1.5.13 and nezha-agent v1.5.6 has been released
https://github.com/nezhahq/nezha/releases/tag/v1.5.13
https://github.com/nezhahq/agent/releases/tag/v1.5.6
:D

Moraxyc commented:
> nezha v1.5.13 and nezha-agent v1.5.6 has been released https://github.com/nezhahq/nezha/releases/tag/v1.5.13 https://github.com/nezhahq/agent/releases/tag/v1.5.6 :D

I'll bump these later.

Moraxyc commented:
themes: #374097

L-Trump commented:
Tested on linux-x86_64 hosts, everything works well.

Moraxyc commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361515`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
    <li>nezha-theme-admin</li>
    <li>nezha-theme-user</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
    <li>nezha-theme-admin</li>
    <li>nezha-theme-user</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
    <li>nezha-theme-admin</li>
    <li>nezha-theme-user</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
    <li>nezha-theme-admin</li>
    <li>nezha-theme-user</li>
  </ul>
</details>

Moraxyc commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361515`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nezha</li>
    <li>nezha-agent</li>
  </ul>
</details>
